src/main.rs

```diff
@@ -494,11 +494,23 @@ impl Toolchain {
     }
 
     fn remove(&self, dl_params: &DownloadParams) -> Result<(), Error> {
-        if !self.is_current_nightly() {
-            eprintln!("uninstalling {}", self);
-            let dir = dl_params.install_dir.join(self.rustup_name());
-            fs::remove_dir_all(&dir)?;
-        }
+        eprintln!("uninstalling {}", self);
+        self.do_remove(dl_params)
+    }
+
+    /// Removes the (previously installed) bisector rustc described by `dl_params`.
+    ///
+    /// The main reason to call this (instead of `fs::remove_dir_all` directly)
+    /// is to guard against deleting state not managed by `cargo-bisect-rustc`.
+    fn do_remove(&self, dl_params: &DownloadParams) -> Result<(), Error> {
+        let rustup_name = self.rustup_name();
+
+        // Guard aginst destroying directories that this tool didn't create.
+        assert!(rustup_name.starts_with("bisector-nightly") ||
+                rustup_name.starts_with("bisector-ci"));
```

Comment:
I understand this idea and seems good but given that we are naming the default installation as nightly https://github.com/rust-lang/cargo-bisect-rustc/pull/55/files#diff-639fbc4ef05b315af92b4d836c31b023R737 I don't see how are we avoiding to call do_remove for the default nightly installations.

Reply:
(hmm unfortunately I just killed that hyperlink via my force push ...)

Reply:
@spastorino can you provide an updated link with the new code I force pushed before seeing your comment? I'm not sure what point you are making and I think having the context will help.

(In particular, when you say "the default installation", I do not know if you are referring to `rustup`'s default installation or to ... `cargo-bisect-rustc's` ?

In any case, `cargo-bisect-rustc` certainly shouldn't be calling `do_remove` for any nightly installations. If we are doing that currently, it sounds like a bug.

Reply:
My comment doesn't seem to apply anymore. Let me review again with the updated code.